test-app/app/src/main/java/com/tns/NativeScriptSyncService.java

```diff
@@ -268,16 +268,21 @@ private void deleteRemovedFiles(File sourceDir, String sourceRootAbsolutePath, S
         if (files != null) {
             for (int i = 0; i < files.length; i++) {
                 File file = files[i];
+                String targetFilePath = file.getAbsolutePath().replace(sourceRootAbsolutePath, targetRootAbsolutePath);
+                File targetFile = new File(targetFilePath);
                 if (file.isFile()) {
                     if (logger.isEnabled()) {
                         logger.write("Syncing removed file: " + file.getAbsolutePath().toString());
                     }
 
-                    String targetFilePath = file.getAbsolutePath().replace(sourceRootAbsolutePath, targetRootAbsolutePath);
-                    File targetFile = new File(targetFilePath);
                     targetFile.delete();
                 } else {
                     deleteRemovedFiles(file, sourceRootAbsolutePath, targetRootAbsolutePath);
+
+                    // this is done so empty folders, if any, are deleted after we're don deleting files.
```

Comment:
The comment is only present in the test-app class. Could you do the same for the build template?